### PR TITLE
fix(agw): Migrate hash list of mmeueid2imsi to protobuf_map

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
Modified code to replace hashtable with protobuf map for mmeueid2imsi hash table in s1ap module
The PR partially addresses the issue, https://github.com/magma/magma/issues/11190

## Test Plan

Executed s1ap sanity test suite and unit test case.
Verified manually with test_attact_detach_with_mme_restart.py test case, to check whether contents of map, mmeueid2imsi are written correctly into DB.

### CODE REVIEW

File: lte/protos/oai/s1ap_state.proto [hunk 1]
```diff
@@ -63,7 +63,7 @@ message S1apState {
 }
 
 message S1apImsiMap {
-  map<uint64, uint64> mme_ue_id_imsi_map = 1; // mme_s1ap_ue_id => IMSI64
```
Comment (ssanadhya) [inline lte/protos/oai/s1ap_state.proto:66]: Please mark this field as deprecated and create a new field for uint32 -> uint64 map.

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp [hunk 2]
```diff
@@ -182,10 +183,11 @@ status_code_e S1apStateManager::read_ue_state_from_db() {
 }
 
 void S1apStateManager::create_s1ap_imsi_map() {
-  s1ap_imsi_map_ = (s1ap_imsi_map_t*)calloc(1, sizeof(s1ap_imsi_map_t));
+  s1ap_imsi_map_ = new s1ap_imsi_map_t();
 
-  s1ap_imsi_map_->mme_ue_id_imsi_htbl =
-      hashtable_uint64_ts_create(max_ues_, nullptr, nullptr);
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp:188]: Can we get rid of the max_ues config after this change?

**Review by ssanadhya — CHANGES_REQUESTED** [review]

**Review by pruthvihebbani — APPROVED** [review]
lgtm

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp (hunk 2 above)
Comment (rsarwad) [reply lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp:188]: Yes! planning to remove as part of porting hash table, state_ue_ht

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
@rsarwad could you regenerate pb file and update your PR? insync check is failing.

**Comment by rsarwad** [comment]
> @rsarwad could you regenerate pb file and update your PR? insync check is failing.

Thanks for reviewing. After correcting pb.go file, insync check is passing

**Comment by ulaskozat** [comment]
@ssanadhya I think this is blocked due to your review request that seems to be addressed.

**Comment by rsarwad** [comment]
> @ssanadhya I think this is blocked due to your review request that seems to be addressed.

The requested comment is addressed at, https://github.com/magma/magma/pull/13335/files#diff-d555416faabf15fcbd20c1014dee7c63c4417cb13d7cd10cf65ba7402d227db7R66
I think @ssanadhya needs to approve the changes

**Review by ssanadhya — APPROVED** [review]
Lgtm

**Pull request merged**